### ISSUE TITLE
Bring reference architecture up to date

### ISSUE THREAD

**Issue by WestDiscGolf** [Azure-Samples/Serverless-microservices-reference-architecture#31]
### This issue is for a: (mark with an `x`)
```
- [ ] bug report -> please search issues before submitting
- [X] feature request
- [X] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce

This reference architecture is really interesting but as with the ever increase in cadence of functionality in Azure it has fallen behind in services, SDK versions, best practices etc. and not been updated for 2 years. It would be really helpful if this reference architecture was updated to work with the latest functionality in Azure Functions etc. and best practices as required.

Thanks!

**Comment by lockhartsoftware** [comment]
Yes please. That’s a excellent idea.

**Comment by joelhulen** [comment]
I would love to update this because I agree on all points. I need to check with the stakeholders, including @gk-8

**Comment by jongio** [comment]
Any update @joelhulen?

**Comment by joelhulen** [comment]
I've not heard back from @gk-8

@jeffhollan Do you have any insight on how we can update this?

I'll send an email to the team to see if we can get this going again. I'm glad to see the interest!

**Comment by lockhartsoftware** [comment]
In my opinion, this is the single best resource that I have discovered on how to design and build a Serverless microservices application using Microsoft Azure. Believe me, I have wasted hours checking out various other resources, and nothing comes close.

I really appreciate the hands-on, detailed walkthrough approach to learning.

Not only do you show the ‘how’, but you also explain the ‘why’.

I have already used this reference architecture in a number of my designs, with great success I might add.

This reference architecture really is an exemplar, and should be the approach taken by Microsoft across all of its technologies.

Please update.

**Comment by jeffhollan** [comment]
Thanks all for flagging - let me circle back with the team and see if we can get someone to spend some cycles here

**Comment by joelhulen** [comment]
Thanks, @jeffhollan! Please feel free to reach out to me if you want some help.

**Comment by lockhartsoftware** [comment]
Could we make the frontend Blazor WASM this time, please?

**Comment by rohannevrikar** [comment]
I completely agree with @lockhartsoftware! I still come back to this repo to understand best practices while using Azure serverless, and I've personally learnt a lot from this resource. I'd love to contribute if things move forward.

**Comment by joelhulen** [comment]
@lockhartsoftware I would actually like to do a WASM version this time around. As much as I love Vue, which I enjoyed using in the current version, I've recently started using Blazor WebAssembly and it's an absolute joy to use. Now I just need the blessing from @jeffhollan to update this architecture 🙂

**Comment by WestDiscGolf** [comment]
A blazor wasm ui sounds like a good plan to me. As the ui is a small part of the architecture then maybe the community can contribute other ui framework versions maybe if its wanted?

**Comment by lockhartsoftware** [comment]
@joelhulen building interactive web UIs using C# instead of JavaScript is, in my opinion, sheer bliss. I spent the last 3-4 years with Angular on the client-side, and now with Blazor WASM I believe that my productivity has spiked significantly.

Oh, go on @jeffhollan, you know it makes sense 🙂

**Comment by lockhartsoftware** [comment]
@joelhulen @jeffhollan Hello! Any update on this? Thanks.

**Comment by jongio** [comment]
Adding @JasRobe to the issue.

**Comment by lockhartsoftware** [comment]
Excellent. Thanks @jongio

**Comment by lockhartsoftware** [comment]
Call me premature, but I guess I got a little excited by the number of people that were added to this thread.

@JasRobe @joelhulen @jeffhollan @gk-8 

Has there been any progress?

Is any work planned?

I remain optimistic!

**Comment by joelhulen** [comment]
Not that I'm aware of. I think we're waiting on @jeffhollan to see how they would like to proceed.

**Comment by WestDiscGolf** [comment]
I appreciate it's a busy time of the release cycle at Microsoft. Can we take from this that .NET 6 / Azure Functions V4 are waiting to be GA'd before this gets looked at? It would make sense to update it to use the next LTS "out-of-process" architecture :)

**Comment by WestDiscGolf** [comment]
Glad to see some "todo" issues raised over night. Looking forward to seeing how this progresses. Thanks @DanielLarsenNZ :-)

**Comment by DanielLarsenNZ** [comment]
Hey team, this sample has been upgraded to Azure Functions v3 / dotnetcore 3.1. This is a stepping stone towards Azure Functions v4 / dotnet 6 support. Work continues!

A release and tag has been created: https://github.com/Azure-Samples/Serverless-microservices-reference-architecture/releases/tag/functions-v3

**Comment by WestDiscGolf** [comment]
Thanks for the update @DanielLarsenNZ. Looking forward to seeing the migration to isolated/out-of-process model. Thanks for all the hard work 😄

**Comment by joelhulen** [comment]
Is there any interest in reviving this reference architecture? I have some cycles now, but I don't want to waste any time if there's no real interest or use for it.

**Comment by lockhartsoftware** [comment]
@joelhulen yes, please. I would be very interested in the revival of this reference architecture.
